clamp.go

```diff
@@ -0,0 +1,7 @@
+package funky
+
+import "golang.org/x/exp/constraints"
```

Comment:
Would be great if you could use cmp.Ordered instead - seems unnecessary to add that dependency :grimacing: